src/gui/painting/qpainter.cpp

```diff
@@ -2673,6 +2673,151 @@ QPainterPath QPainter::clipPath() const
     }
 }
 
+static QPainterPath noClipping(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return QPainterPath();
+}
+
+static QPainterPath PathClipReplaceClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return info.path * matrix;
+}
+
+static QPainterPath PathClipIntersectClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return PathClipReplaceClip(matrix, info);
+}
+
+static QPainterPath RectClipReplaceClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    QPainterPath tempPath;
+    tempPath.addRect(info.rect);
+    tempPath = tempPath * matrix;
+    return tempPath;
+}
+
+static QPainterPath RectClipIntersectClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return RectClipReplaceClip(matrix, info);
+}
+
+static QPainterPath RectFClipReplaceClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    QPainterPath tempPath;
+    tempPath.addRect(info.rectf);
```

Comment:
How come this RectF one does not do tempPath = tempPath * matrix like the integer Rect one does.  Possibly needs a comment noting the reason.